code/game/objects/structures/plasticflaps.dm

```diff
@@ -57,15 +57,11 @@
 	name = "airtight plastic flaps"
 	desc = "Heavy duty, airtight, plastic flaps."
 
-/obj/structure/plasticflaps/mining/New() //set the turf below the flaps to block air
-	var/turf/T = get_turf(loc)
-	if(T)
-		T.blocks_air = 1
-	..()
+/obj/structure/plasticflaps/mining/New()
+	air_update_turf(1)
+	
+/obj/structure/plasticflaps/mining/CanAtmosPass()
+	return FALSE
```

Comment:
The return value needs to be set to TRUE in Destroy(), before air_update_turf() is called